app/experimenter/static/js/components/PrefForm.js

```diff
@@ -15,72 +17,102 @@ import {
 export default class PrefForm extends React.PureComponent {
   static propTypes = {
     data: PropTypes.instanceOf(Map),
+    type: PropTypes.string,
     errors: PropTypes.instanceOf(Map),
     handleDataChange: PropTypes.func,
     handleErrorsChange: PropTypes.func,
   };
 
+  renderSingularPrefInfo() {
+    if (!this.props.data.get("is_multi_pref")) {
+      return (
+        <div>
+          <DesignInput
+            label="Pref Name"
+            name="pref_key"
+            id="id_pref_key"
+            onChange={value => {
+              this.props.handleDataChange("pref_key", value);
+            }}
+            value={this.props.data.get("pref_key")}
+            error={this.props.errors.get("pref_key", "")}
+            helpContent={PREF_KEY_HELP}
+          />
+
+          <DesignInput
+            label="Pref Type"
+            name="pref_type"
+            id="id_pref_type"
+            onChange={value => {
+              this.props.handleDataChange("pref_type", value);
+            }}
+            value={this.props.data.get("pref_type")}
+            error={this.props.errors.get("pref_type", "")}
+            as="select"
+            helpContent={PREF_TYPE_HELP}
+          >
+            <option>Firefox Pref Type</option>
+            <option>boolean</option>
+            <option>integer</option>
+            <option>string</option>
+            <option>json string</option>
+          </DesignInput>
+
+          <DesignInput
+            label="Pref Branch"
+            name="pref_branch"
+            id="id_pref_branch"
+            onChange={value => {
+              this.props.handleDataChange("pref_branch", value);
+            }}
+            value={this.props.data.get("pref_branch")}
+            error={this.props.errors.get("pref_branch", "")}
+            as="select"
+            helpContent={PREF_BRANCH_HELP}
+          >
+            <option>Firefox Pref Branch</option>
+            <option>default</option>
+            <option>user</option>
+          </DesignInput>
```

Comment:
We've done this a couple times in the react so far, where we omit a value from `<option>` and the label just *happens* to be the actual correct value, but we shouldn't keep that, I cleaned up a bunch of them in my PR but.... you know we should avoid it.